### PR TITLE
ISPN-9398 & ISPN-9427 Hotrod configuration via properties file

### PR DESCRIPTION
**Pull request by karesti**
Completes 
https://issues.jboss.org/browse/ISPN-9398
https://issues.jboss.org/browse/ISPN-9427

### CODE REVIEW

**Comment by karesti** [comment]
@tristantarrant I added the xsite documentation and a missing property for transactions

File: client/hotrod-client/src/main/java/org/infinispan/client/hotrod/configuration/package-info.java [hunk 1]
```diff
@@ -342,6 +342,19 @@
...
+ *          <th colspan="4">XSite properties</th>
+ *       </tr>
+ *       <tr>
+ *          <td><b>infinispan.client.hotrod.cluster.SITE</b></td>
+ *          <td>String HOST and int PORT configuration</td>
+ *          <td>Example for siteA and siteB:<br/>
+ *           infinispan.client.hotrod.cluster.siteA=hostA1:11222; hostA2:11223`<br/>
+ *           infinispan.client.hotrod.cluster.siteB=hostB1:11222; hostB2:11223`
+ *          </td>
+ *          <td>The default near-cache {@link org.infinispan.client.hotrod.configuration.ClusterConfigurationBuilder#addCluster(java.lang.String)} and
```
Comment (tristantarrant) [inline client/hotrod-client/src/main/java/org/infinispan/client/hotrod/configuration/package-info.java]: near-cache ?

**Comment by karesti** [comment]
@tristantarrant I made the modifications and added some extra property setters

**Comment by karesti** [comment]
test failures unrelated

File: client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/ConfigurationProperties.java [hunk 2]
```diff
@@ -385,6 +397,10 @@ public void setTransactionMode(String transactionMode) {
       props.setProperty(TRANSACTION_MODE, transactionMode);
    }
 
+   public void setTransactionTimeout(String transactionTimeout) {
```
Comment (pruivo) [inline client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/ConfigurationProperties.java]: I guess it can be a `long` instead of `string`?

**Comment by karesti** [comment]
@pruivo modified to long and squashed commits

**Pull request merged**